src/main/java/org/springframework/data/couchbase/core/ExecutableFindByIdOperation.java

```diff
@@ -46,28 +46,29 @@
 
 	}
 
-	interface FindByIdWithCollection<T> extends TerminatingFindById<T> {
+	interface FindByIdWithCollection<T> {
 
 		/**
 		 * Allows to specify a different collection than the default one configured.
 		 *
 		 * @param collection the collection to use in this scope.
 		 */
-		TerminatingFindById<T> inCollection(String collection);
+		ExecutableFindById<T> inCollection(String collection);
 
```

Comment:
if each with-er returns ExecutableFindById, then there is no need to chain the interfaces.